### PR TITLE
Refactor publishing API presenters (part 1)

### PR DESCRIPTION
**Pull request by mobaig**
This is a first pass at refactoring the publishing API presenters in Whitehall for greater clarity and to make changes easier. In the process, we also resolve some buggy and redundant behaviour. It will be followed by a 2nd PR that will attempt to simplify the inheritance hierarchy in these presenters.

Specifically, this PR:

- Moves the resolution of `topics` and `parent` link types into the LinksPresenter. This provides a single place for resolving these tag types for a given item. For example, any children of the EditionsPresenter will now have access to this logic via the `extract_links` methods.
- Splits apart the PlaceholderPresenter - more details in 73e0b00. This was unnecessarily confusing and made it harder to set specific link types for each of the four items it was presenting.
- Adds the 'parent' link to children of the EditionPresenter - this was missed and should be sent with each links update.

### CODE REVIEW

**Comment by jackscotti** [comment]
👍  looks great

File: app/presenters/publishing_api_presenters/ministerial_role.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require_relative "../publishing_api_presenters"
+
+class PublishingApiPresenters::MinisterialRole < PublishingApiPresenters::Item
```
Comment (boffbowsh) [inline app/presenters/publishing_api_presenters/ministerial_role.rb]: I understand the need to demystify the code path for presenting these, however I don't think this duplication of code is helpful. I think these need to either subclass `Placeholder`, or the common methods need to be extracted into a module and mixed in.

**Comment by boffbowsh** [comment]
Thanks for this, apart from my comment it looks great and is much more readable. I'm just concerned that by duplicating this much code, someone will end up forgetting to update a method in one of those classes and we won't notice. I think a mixin approach would be great here for defining standard base functionality.

File: app/presenters/publishing_api_presenters/ministerial_role.rb (hunk 1 above)
Comment (mobaig) [inline app/presenters/publishing_api_presenters/ministerial_role.rb]: Completely understand the concern around duplication. This is very much an intermediate state that we're looking to resolve in our next PR. We've refrained from extracting out this duplication thus far so we can get a better handle on the kinds of repetition across all of the presenters. I think this'll let us make a more informed choice about the correct abstraction to use when sharing code across the classes, whether that's mixins or a small set of composable objects.

Comment (boffbowsh) [inline app/presenters/publishing_api_presenters/ministerial_role.rb]: Ok, that sounds good to me :)

**Comment by boffbowsh** [comment]
👍 from me

**Pull request merged**